src/routes/docs/products/ai/integrations/perplexity/+page.markdoc

```diff
@@ -13,4 +13,310 @@ Integrating Perplexity into your Appwrite project is simple. This tutorial will
+ */
+export function getStaticFile(fileName) {
+  return fs.readFileSync(path.join(staticFolder, fileName)).toString();
+}
+```
+
+By separating the `getStaticFile()` function into a `utils.js` file, we can easily reuse this function in other parts of our function if we extend on top of this.
+
+While we're at it, we're also going to create a `throwIfMissing()` function in the same file which we'll use later to validate body and environment variables:
+
+```js
+/**
+ * Throws an error if any of the keys are missing from the object
+ * @param {*} obj
+ * @param {string[]} keys
+ * @throws {Error}
+ */
+export function throwIfMissing(obj, keys) {
+  const missing = [];
+  for (let key of keys) {
+    if (!(key in obj) || !obj[key]) {
+      missing.push(key);
+    }
+  }
+  if (missing.length > 0) {
+    throw new Error(`Missing required fields: ${missing.join(', ')}`);
+  }
+}
+```
+
+{% /section %}
+
+{% section #step-4 step=4 title="Create HTML File" %}
+Next, we'll create the static HTML file that we'll return. First we're going to create some HTML Boilerplate in `static/index.html`:
+
+```html
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Perplexity Demo</title>
+  </head>
+  <body></body>
+</html>
+```
+
+To keep this tutorial simple, we're going to use [AlpineJS](https://alpinejs.dev/) as our JS Framework of choice.
+To do this we're going to a new script tag to our html file's head tag:
+
+```html
+<script src="//unpkg.com/alpinejs" defer></script>
+```
+
+Finally to provide some styling we're going to add use [Pink Design](https://pink.appwrite.io), our CSS Framework. Feel free to use any CSS Framework you like though.
+
+Add the following link tags to your HTML file's head tag:
+
+```html
+<link rel="stylesheet" href="https://unpkg.com/@appwrite.io/pink" />
+<link rel="stylesheet" href="https://unpkg.com/@appwrite.io/pink-icons" />
+```
+
+This completes the boilerplate for our HTML file, we'll continue to add the rest of the HTML and AlpineJS code in the next steps.
+{% /section %}
+
+{% section #step-5 step=5 title="Serving HTML" %}
+Now that we have our HTML file, we need to serve it from our function. To do this we'll add the following code to our `main.js` file:
+
+```js
+import { getStaticFile, throwIfMissing } from './utils.js';
+
+export default async ({ req, res, error }) => {
+  throwIfMissing(process.env, ['PERPLEXITY_API_KEY']);
+
+  if (req.method === 'GET') {
+    return res.send(getStaticFile('index.html'), 200, {
+      'Content-Type': 'text/html; charset=utf-8',
+    });
+  }
+};
+```
+
+Notice we also added a check to ensure that the `PERPLEXITY_API_KEY` environment variable is set. This is the API key we'll use to authenticate with the Perplexity API.
+{% /section %}
+
+{% section #step-6 step=6 title="Use the Perplexity API" %}
```

Comment:
Step 4, 5, 7 should just be a single step. This is too many irrelevant steps, also confusing to go form handling GET requests -> POST -> back to get.

Reply:
I'm just concerned it'll be a huge step, I don't want them just to build this I want them to understand what they are doing while building it